### PR TITLE
Improve regexes in docs

### PR DESCRIPTION
**Pull request by scop**
Remove unneeded capturing group parenthesis as there's no significance to the groups, misc strictness fixes, make do the right thing with both match and search style regex matching.

### CODE REVIEW

File: qa/samples/config/ignore-release-commits [hunk 1]
```diff
@@ -1,7 +1,7 @@
 [ignore-by-title]
-regex=^Release(.*)
+regex=^Release.*
 ignore=T5,T3
 
 [ignore-by-body]
-regex=(.*)relëase(.*)
```
Comment (scop) [inline qa/samples/config/ignore-release-commits:6]: I'm assuming the `ë` (vs plain`e`)  was not intended here, but did not check.

**Comment by jorisroovers** [comment]
Thanks for the PRs. I currently have no cycles to work on gitlint, so it will take me a while to get to them (perhaps more than a month)... But I will eventually (or @sigmavirus24)

File: qa/samples/config/ignore-release-commits (hunk 1 above)
Comment (jorisroovers) [reply qa/samples/config/ignore-release-commits:6]: It is intentional, to check that unicode works properly. Mostly needed back when we were supporting python 2, but there's still edge cases for which I think this is useful so I tend to include non-ascii chars in the test data.

Comment (scop) [reply qa/samples/config/ignore-release-commits:6]: Ok, modified and rebased to keep it.
